lib/class-wp-json-base-posts-controller.php

```diff
@@ -68,6 +70,8 @@ public function get_item( $request ) {
 			return new WP_Error( 'json_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
 		}
 
+		setup_postdata( $post );
```

Comment:
If we set this up here, we need to ensure we restore it afterwards. We're already doing that inside `prepare_item_for_response`, do we need it here too?